internal/component/prometheus/remotewrite/remote_write_test.go

```diff
@@ -80,11 +81,20 @@ func Test(t *testing.T) {
 		},
 	}}
 
+	var results []prompb.TimeSeries
 	select {
 	case <-time.After(time.Minute):
 		require.FailNow(t, "timed out waiting for metrics")
 	case res := <-writeResult:
-		require.Equal(t, expect, res.Timeseries)
+		if len(res.Timeseries) == 1 {
+			results = append(results, res.Timeseries[0])
+		} else if len(res.Timeseries) == 2 {
+			results = res.Timeseries
+		}
+		// When we have two results make sure they match what we expect
+		if len(res.Timeseries) == 2 {
+			require.Equal(t, expect, res.Timeseries)
+		}
```

Comment:
Drive-by: This test failed so I thought I broke it but it's just flaky
```
go test ./internal/component/prometheus/remotewrite/... -run '^Test$' -count 20
```
used to break it locally